flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -99,7 +106,7 @@ else if (old instanceof ConnectingChannel) {
 
 			// Make sure to increment the reference count before handing a client
 			// out to ensure correct bookkeeping for channel closing.
-			if (!client.incrementReferenceCounter()) {
+			if (client != null && !client.incrementReferenceCounter()) {
```

Comment:
Does the `client` have the possibility to be still `null` here?

Reply:
No need. Mistakenly added.